src/rpc/blockchain.cpp

```diff
@@ -1981,10 +1981,104 @@ UniValue savemempool(const UniValue &params, bool fHelp)
     return NullUniValue;
 }
 
+UniValue getchaintxstats(const UniValue &params, bool fHelp)
+{
+    if (fHelp || params.size() > 2)
+    {
+        throw std::runtime_error(
+            "getchaintxstats ( nblocks blockhash )\n"
+            "\nCompute statistics about the total number and rate of transactions in the chain.\n"
+            "\nArguments:\n"
+            "1. nblocks      (numeric, optional) Size of the window in number of blocks (default: one month).\n"
+            "2. \"blockhash\"  (string, optional) The hash of the block that ends the window.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"time\": xxxxx,        (numeric) The timestamp for the statistics in UNIX format.\n"
+            "  \"window_final_block_hash\": \"...\",      (string) The hash of the final block in the window.\n"
+            "  \"window_final_block_height\": xxxxx,    (numeric) The height of the final block in the window.\n"
+            "  \"window_block_count\": xxxxx,           (numeric) Size of the window in number of blocks.\n"
+            "  \"window_tx_count\": xxxxx,              (numeric) The number of transactions in the window. Only "
+            "returned if \"window_block_count\" is > 0.\n"
+            "  \"window_interval\": xxxxx,              (numeric) The elapsed time in the window in seconds. Only "
+            "returned if \"window_block_count\" is > 0.\n"
+            "  \"txcount\": xxxxx,     (numeric) The total number of transactions in the chain up to that point.\n"
+            "  \"txrate\": x.xx,       (numeric) The average rate of transactions per second in the window.\n"
+            "}\n"
+            "\nExamples:\n" +
+            HelpExampleCli("getchaintxstats", "") + HelpExampleRpc("getchaintxstats", "2016"));
+    }
+    const CBlockIndex *pindex;
+    int blockcount = 30 * 24 * 60 * 60 / Params().GetConsensus().nPowTargetSpacing; // By default: 1 month
+
+    if (params.size() > 0 && !params[0].isNull())
+    {
+        blockcount = params[0].get_int();
+    }
+
+    bool havehash = params.size() > 1 && !params[1].isNull();
+    uint256 hash;
+    if (havehash)
+    {
+        hash = uint256S(params[1].get_str());
+    }
+
+    {
+        LOCK(cs_main);
+        if (havehash)
+        {
+            auto it = mapBlockIndex.find(hash);
+            if (it == mapBlockIndex.end())
+            {
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
+            }
+            pindex = it->second;
+            if (!chainActive.Contains(pindex))
+            {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Block is not in main chain");
+            }
+        }
+        else
+        {
+            pindex = chainActive.Tip();
+        }
+    }
+
+    DbgAssert(pindex != nullptr, );
```

Comment:
since you subsequently use pindex->nHeight, this DbgAssert should implement a release mode action that will stop the core dump such as:
DbgAssert(pindex != nullptr, throw JSONRPCError(...) );

Reply:
right will fix